feat(replays): Update the Replay Index page columns and Details page header to match

ryan953 opened the pull request:
New css in the app to match the [figma](https://www.figma.com/file/fOXcP5EaaTnerWqrydnyCP/Specs%3A-Index-v3?node-id=190-38289&t=swXsyTNh1ysgRSI6-0):




### Table rows updated:
![SCR-20230316-j17](https://user-images.githubusercontent.com/187460/225747971-e8568a0a-4adc-4c3b-8d87-36b947794ce5.png)


### Details page header updated:
| With error count | No errors | Loading
| --- | --- | --- |
| ![SCR-20230316-j18](https://user-images.githubusercontent.com/187460/225747988-e5b654f4-feaa-47d1-a87b-2fe528ca1daf.png)  | ![SCR-20230316-kn5](https://user-images.githubusercontent.com/187460/225760571-e0fa6349-acc0-4bce-b29e-babc534c9c3e.png) | ![SCR-20230316-kmu](https://user-images.githubusercontent.com/187460/225760593-d2b497f8-f655-4120-bb00-735a9a3040a3.png) |


Here's the diff of the header loading state and the header with info:
![SCR-20230316-kni](https://user-images.githubusercontent.com/187460/225760618-cc0033cf-e8c0-4548-b45c-0a9f8f5ddafa.png)





Fixes #42784
Fixes #44759

ryan953 committed:
wip

billyvg commented:
![image](https://user-images.githubusercontent.com/79684/225466304-544e1b4e-574d-47c3-a2af-d18ff189189f.png)
So many numbers, I wonder how important the version numbers are? What if we moved the versionto a tooltip?

billyvg commented on static/app/components/replays/contextIcon.tsx:
```diff
@@ -0,0 +1,37 @@
+  return (
+    <div className={className}>
+      <Tooltip title={name}>
+        <Suspense fallback={<LoadingMask />}>
```
Will the mask have a proper size?

ryan953 replied:
yes somehow!

i'll get a screen shot of the loading state

billyvg commented on static/app/views/replays/detail/page.tsx:
```diff
@@ -33,12 +34,28 @@ function Page({children, crumbs, orgSlug, replayRecord}: Props) {
+            email: replayRecord.user.email || '',
+            id: replayRecord.user.id || '',
+            ip_address: replayRecord.user.ip || '',
+            name: replayRecord.user.username || '',
```
Has this been fixed? I remember a support ticket where user was trying to search via username, but it was being aliased to `name` on the BE.

ryan953 replied:
i'll check what the backend has and get that fixed. 

We should have `user.username` on the ReplayRecord, per the docs: https://docs.sentry.io/product/sentry-basics/search/searchable-properties/

This UserBadge component accepts a few variations, i think we'll have to keep this mapping the way it is.

Jesse-Box commented:
@ryan953 looking good mate, just a few things I can see which are different from my designs
1. The user avatars are a little big. It should be 24px in size
2. The project badge sitting to the left of the replay ID is a tad too big. Looks to be about 16px, when it should be 12px in size.
3. The font-size of the error count (Details Page) is smaller than, for example, the start time
4. The user identifier string should be bold/medium font-weight.

ryan953 committed:
code review comments

ryan953 commented:
@Jesse-Box  following up:

1. The user avatars are a little big. It should be 24px in size
   - updated the avatar on the list page to be 24px
   - avatar on the details page is still 32px
3. The project badge sitting to the left of the replay ID is a tad too big. Looks to be about 16px, when it should be 12px in size.
  - updated badge on list page to be 12px
  - badge on details page (see `Session Replay > $BADGE $REPLAY_ID (NEW)`) is still 16px
5. The font-size of the error count (Details Page) is smaller than, for example, the start time
  - fixed, was missing the `<KeyMetricData>` wrapper
7. The user identifier string should be bold/medium font-weight.
  - fixed, using `<Layout.Title>` instead of a new styled component (I left a todo for this, it's to-done now)

ryan953 committed:
fixup placeholders in loading state

ryan953 committed:
fix tests

ryan953 committed:
Merge branch 'master' into ryan953/44759-replay-os-browser-cols

billyvg commented on static/app/views/issueDetails/groupReplays/groupReplays.tsx:
```diff
@@ -19,9 +17,16 @@ type Props = {
+
 function GroupReplays({group}: Props) {
-  const theme = useTheme();
-  const hasRoomForColumns = useMedia(`(min-width: ${theme.breakpoints.small})`);
```
How does this look at small breakpoints now?

ryan953 replied:
I'll take some screen shots!

billyvg commented on static/app/views/replays/replayTable/tableCell.tsx:
```diff
@@ -30,43 +34,64 @@ export function SessionCell({
+        displayName={
+          <MainLink to={replayDetails}>{replay.user.display_name || ''}</MainLink>
+        }
+        user={{
+          username: replay.user.display_name || '',
+          email: replay.user.email || '',
+          id: replay.user.id || '',
+          ip_address: replay.user.ip || '',
+          name: replay.user.username || '',
+        }}
```
I think we do this in at least one other place... might be worth refactoring into a util fn.

billyvg commented:
Some additional design grumblings... The avatars here introduce a lot of vertical whitespace 
![image](https://user-images.githubusercontent.com/79684/226411245-df646fce-044e-4951-9a1c-3cf5a926ec25.png)


They also encompass an additional line of information now as it's centered between user, url walker, and now project + timestamp.


The layout here is a bit broken too:
![image](https://user-images.githubusercontent.com/79684/226412107-cc7d0e21-1535-4151-b079-e4888447e7c3.png)
I would expect it to take up the full width and display the entire URL instead of truncating.

billyvg commented:
Oh, this is a blocker: the project is wrong, looks like it's showing a short id for the replay? Or is that intended?

billyvg committed:
feat replay fix url walker width (#46066)

StringWalker was not using the full width when available.


Replay Details:

![image](https://user-images.githubusercontent.com/79684/226421784-3d4361ca-834b-4053-80bd-9afbf1ade955.png)

Replay Index:

![image](https://user-images.githubusercontent.com/79684/226421848-9bd5caef-9ac9-453c-b83a-345fdddd7443.png)

billyvg commented:
Ah noticed a case I missed re: url walker..., if it's exactly 3 items the middle will not shrink

ryan953 commented:
> Oh, this is a blocker: the project is wrong, looks like it's showing a short id for the replay? Or is that intended? (If it's intended then I find that very confusing).

Yes. We've got the Project Icon + Replay-shortID
I think the project icon is not super important to have, and the replay_id is something new that we're showing... it's in the tables around the app already (discover, issue>replays, etc) so we need it in this table, in the details page header someplace. I think we'll put it in and see how that goes. It could be something to lift above the username/ip address too

ryan953 committed:
remove OS & Browser version info at small sizes, improve details header layout at small sizes
